hysds_commons/stac_utils.py

```diff
@@ -0,0 +1,106 @@
+import copy
+import os
+import json
+from hysds_commons.log_utils import logger
+
+def generate_bbox(coord_list):
+# This function creates a bounding box for an input list of coordinates describing a polygon.  The function is called within the create_stac_doc function.
+    '''
+    :param coord_list (dict): a list of coordinates where each coordinate is expressed as x, y
+    :return ret (list): the bbox created for the input polygon
+    '''
+    box = []
+    for i in (0, 1):
```

Comment:
i'm still confused by the way we're sorting the coordinates
what is the significance of sorting the coordinates this way? will each coordinate only consist of 2 points or can it consist of more points? `ex. [1,2] vs. [1,2,...,n]`

why not simply flatten the list and sort it?
ex. `[[1,2],[3,4]] -> [1,2,3,4]`

Reply:
im playing around with the function and it looks like the for loop may be redundant b/c the results are the same each iteration
```python
def generate_bbox(coord_list):
    '''
    This function creates a bounding box for an input list of coordinates describing a polygon.
    The function is called within the create_stac_doc function.
    :param coord_list (dict): a list of coordinates where each coordinate is expressed as x, y
    :return ret (list): the bbox created for the input polygon
    '''
    box = []
    print("input", coord_list)
    for i in (0, 1):
        res = sorted(coord_list, key=lambda x: x[i])
        print(i, res)
        box.append((res[0][i], res[-1][i]))
    ret = f"({box[0][0]}, {box[1][0]}, {box[0][1]}, {box[1][1]})"
    print(ret)
    print("")
    return ret

generate_bbox([[1,2],[3,4]])
# input [[1, 2], [3, 4]]
# 0 [[1, 2], [3, 4]]
# 1 [[1, 2], [3, 4]]
# (1, 2, 3, 4)

generate_bbox([[2,1],[4,3]])
# input [[2, 1], [4, 3]]
# 0 [[2, 1], [4, 3]]
# 1 [[2, 1], [4, 3]]
# (2, 1, 4, 3)

generate_bbox([[2,1],[3,4]])
# input [[2, 1], [3, 4]]
# 0 [[2, 1], [3, 4]]
# 1 [[2, 1], [3, 4]]
# (2, 1, 3, 4)

generate_bbox([[1,2],[4,3]])
# input [[1, 2], [4, 3]]
# 0 [[1, 2], [4, 3]]
# 1 [[1, 2], [4, 3]]
# (1, 2, 4, 3)

generate_bbox([[4,3],[2,1]])
# input [[4, 3], [2, 1]]
# 0 [[2, 1], [4, 3]]
# 1 [[2, 1], [4, 3]]
# (2, 1, 4, 3)

generate_bbox([[4,3],[1,2]])
# input [[4, 3], [1, 2]]
# 0 [[1, 2], [4, 3]]
# 1 [[1, 2], [4, 3]]
# (1, 2, 4, 3)

generate_bbox([[3,4],[1,2]])
# input [[3, 4], [1, 2]]
# 0 [[1, 2], [3, 4]]
# 1 [[1, 2], [3, 4]]
# (1, 2, 3, 4)

generate_bbox([[3,4],[2,1]])
input [[3, 4], [2, 1]]
0 [[2, 1], [3, 4]]
1 [[2, 1], [3, 4]]
(2, 1, 3, 4)

```

Reply:
@kynjonez @DustinKLo: According to the spec here:

https://github.com/radiantearth/stac-spec/blob/master/item-spec/item-spec.md#item-fields

it looks like the `bbox` field (minimum bounding rectangle) is required if we have an GeoJSON defined and the bbox array is: `[westlon, minlat, eastlon, maxlat]`.

We need to be able to handle all sorts of edge cases, e.g. a polygon that crosses the antimeridian/international dateline, so a more complete solution to extract the bbox would be to use the `shapely` library. Here is some example code from nisar-pcm that does exactly that: https://github.jpl.nasa.gov/IEMS-SDS/nisar-pcm/blob/734df02da122e4a6adf11511f58ecd9bee25e71e/nisar_chimera/precondition_functions.py#L1398-L1406

You could do the same thing in your `generate_bbox`. Hope this helps.

Reply:
@kynjonez: Alternatively we want to minimize 3rd party library dependencies, you can use one of the lighter-weight GeoJSON python libraries to extract the bbox since the bbox is part of the GeoJSON spec.

Reply:
Code has been updated to use the Shapely library and has been tested on factotum as well. I also updated setup.py to include shapely.

@pymonger @DustinKLo Please review the latest code and let me know if there are any concerns or other changes I may need to make.

Reply:
@kynjonez: looks good.